src/Runner.Worker/StepsRunner.cs

```diff
@@ -352,7 +364,70 @@ private void CompleteStep(IStep step, TaskResult? result = null, string resultCo
         {
             var executionContext = step.ExecutionContext;
 
+            if (ShouldUploadAttachment(step))
+            {
+                QueueStepSummaryUpload(step);
+            }
+
             executionContext.Complete(result, resultCode: resultCode);
         }
+
+        private bool ShouldUploadAttachment(IStep step)
+        {
+          var executionContext = step.ExecutionContext;
+          var stepSummaryFilePath = executionContext.GetGitHubContext("step_summary");
+
+          if(!ActionCommandManager.StepSummaryEnabled(step.ExecutionContext))
+          {
+              Trace.Info("Step Summary is disabled; skipping attachment upload");
+              return false;
+          }
+
+          var summaryExists = File.Exists(stepSummaryFilePath);
+          if (summaryExists)
```

Comment:
nit, consider `if (File.Exists(stepSummaryFilePath))`

Reply:
np! fixed in 901a406